test/distributed/_sharded_tensor/ops/test_embedding.py

```diff
@@ -51,20 +76,52 @@ def _run_sharded_embedding(self, spec, input_size, num_embeddings, embedding_dim
         inp = torch.randint(0, num_embeddings, tuple(input_size)).cuda(self.rank)
         sharded_output = sharded_embedding(inp)
 
+        # If max_norm is set, we need to ensure that the renorm has been applied across
+        # inputs from all ranks.
+        if max_norm is not None:
+            gathered_inputs = [torch.zeros_like(inp) for _ in range(TEST_GPU_NUM)]
+            dist.all_gather(gathered_inputs, inp)
+            unique_inp = torch.unique(torch.cat(gathered_inputs))
+            local_embedding(unique_inp)
+
         # Run local computation
         local_output = local_embedding(inp)
 
+        # Compare local weight and shared one to ensure the renorm
+        # as expected.
+        if max_norm is not None:
```

Comment:
Are we testing cases here where the inputs to the embedding table are a subset of the total embeddings? If so, I'm wondering why this test doesn't fail since in sharded case update the entire embedding table but in local we only update the looked up indices. Unless I'm missing something here.

Reply:
To answer your question:

1. If we don't compare the weight, we will not know the difference because the results are not aware of whether we do renorm globally or to the looked up indices only. Because if we don't look up these indices, the weight change for these indices are not tested. 
2. I gather all inputs from all ranks and send them to each local first, so that shared weights are the same as the local one. And our test cases do cover that. If let's say we don't sync the input across all ranks, the test does fail.


Does this answer your question?